### PR TITLE
Tier 4 box 4a — the citation relation as real structure, without its resolution

### PR DESCRIPTION
**Pull request by justinlooney**
`WM_SCOPE.md` §7 made `Explain` wait on *"derivation edges being real structure rather than a JSON array of identifiers"*. This is that structure — and the ruling it lands under is mostly about what it is **forbidden to contain**.

## `KIRRA-WM-PROVENANCE-GRAPH-001`

> Materialize the citation relation at append; **do not materialize its resolved target.**

The real edge is `source event → cites observation id X`, never `source event → target event row Y`. At generation *T*, X may resolve to exactly one row, several, or none:

- `world_events.observation_id` is indexed but **not unique** — many-to-many, not a parent pointer;
- nothing requires a cited id to exist at all;
- a citation dangling at *T* may become resolvable later, and Tier 4 must still report it as dangling **then**.

Baking a target in here would fix, at write time, an answer whose only correct value depends on the coordinate the question is asked at. That is §7's own warning — *"2d's trap re-appearing one tier up, and it will look like reuse when it arrives"* — foreclosed structurally rather than by discipline: `provenance_edges` is keyed `(source_generation, ordinal) → cited_observation_id`, with no target column and nowhere to put one without a migration a reviewer would see.

Order and duplicates are preserved. A source citing the same observation twice said so twice, and deduplicating would index a provenance array the hash does not cover.

## A deterministic index, never independent evidence

Two consequences that are not optional:

**Compaction deletes an event's edges with the event**, in the same transaction. An edge outliving its source would be the index promoting itself to evidence — a citation still readable after the hash-covered statement it came from was deleted, with nothing left to check it against. The branch reads `Degraded`, which is the true answer; the edges are not what carries it.

**An empty edge set is ambiguous, so coverage is recorded.** It is what a source citing nothing looks like, *and* what an un-backfilled store makes every source look like — a positive claim about provenance, made about the whole log, silently. A `provenance_edges_floor` meta row records the highest generation the index does not cover.

This one was found by checking a migration comment I had just written, which claimed the two cases were already distinguishable by whether the source event was retained. They are not: an un-backfilled store has a **retained** source with zero edges. It is Tier 3 case 8's absent-because-unknown / absent-because-empty distinction, at the storage layer.

## Boundedness — the gate earned its keep

Nothing bounds how many observations an event may cite: not the schema, not the domain, not any argument. So the whole-array read was structurally unbounded *while looking fine at the API* — the exact shape of the three defects box 3d exists for. The gate caught it as an unclassified public method rather than as a review comment.

Classifying it `unbounded` and leaving 4b to bound it would have been the "move the unbounded call behind another helper" move ruled out in 3d. Instead `citations_of` takes a limit, refuses `0` and over-ceiling rather than clamping, and reports truncation from a probe row rather than inferring it from `len() == limit` — which is ambiguous exactly at the boundary, where a source with precisely `limit` citations is complete.

## Mutations

Each isolates cleanly:

| Mutation | Reds |
|---|---|
| drop the compaction edge delete | 2 — both compaction tests |
| `BEGIN`/`COMMIT` instead of `SAVEPOINT` | 1 — the nested adjudication write |
| index only citations that resolve today | 6 — including the decisive one |
| absent floor reads as full coverage | 1 — the fail-closed default |

The third also exposed a vacuous assertion of my own: the plural test compared `before` to `after` without pinning that `before` was non-empty, so under that mutation `[] == []` passed. Fixed, and re-verified that it now bites.

## Two things recorded as rejected rather than left as folklore

- The backfill **replaces** rather than merges — but not for the reason that first suggested itself. A merge cannot in fact interleave ordinals from a half-written attempt: the log is immutable and both writers are atomic. It replaces because its postcondition must depend on the source events **alone**, which is what makes the table an index rather than evidence.
- The edge write uses a `SAVEPOINT`, not `BEGIN`: `append_adjudication` already holds a transaction across its call to `append`, and SQLite refuses a nested `BEGIN`. One test covers exactly that, and exactly that test reds when the primitive is swapped.

## Scope also recorded

§7 now carries the 4a/4b/4c split, the audit of what 3f did and did not take off this tier, the three first-class resolution outcomes (`Resolved` / `Plural` / `Dangling`), **cycles as an outcome distinct from truncation**, and the T1/T2 historical-honesty test 4b must satisfy.

## Verification

- 14 new tests in `citation_edges.rs`; **39 suites green** across the world crates
- `cargo test --workspace` green; `cargo fmt --all --check` and `cargo clippy --all-targets` clean
- all six world CI gates green, including the boundedness gate at zero violations

---
_Generated by [Claude Code](https://claude.ai/code/session_01XMCmUGL26v2Hk6eufnShxW)_

### CODE REVIEW

File: crates/kirra-world-store/src/lib.rs [hunk 1]
```diff
@@ -1154,6 +1183,41 @@ impl WorldStore {
...
+        // than a per-append write.
+        if from < 7 {
+            tx.execute_batch(schema::SCHEMA_V7_MIGRATION)?;
+            let head: i64 = tx
+                .query_row(
+                    "SELECT COALESCE(MAX(generation), 0) FROM world_events",
+                    [],
+                    |r| r.get(0),
+                )
+                .unwrap_or(0);
```
[the remark this answers is not in the document]
Comment (justinlooney) [reply crates/kirra-world-store/src/lib.rs]: Correct, and the direction of the failure is the part that makes it serious. Fixed in dba837e1.

`COALESCE(MAX(generation), 0)` already returns a row for an empty table, so the `unwrap_or(0)` could never have been covering the no-rows case — the only thing it could swallow was a genuine error. And it mapped that error to `0`, which is the value meaning *fully covered*. A store that could not read its own head would have claimed to have indexed every citation in it.

That inverts the one property the floor exists to provide. Now propagates with `?`, with the reasoning recorded at the site so the default does not come back as a tidy-up.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/kirra-world-store/src/lib.rs [hunk 2]
```diff
@@ -1394,9 +1458,101 @@ impl WorldStore {
                 e.subject_ref.map(SubjectRef::as_str),
             ],
         )?;
+        self.write_citation_edges(generation, e.provenance)?;
         Ok(generation)
```
[the remark this answers is not in the document]
Comment (justinlooney) [reply crates/kirra-world-store/src/lib.rs:1505]: This one is right, and it is the sharpest finding of the three: the doc comment directly above the savepoint *asserted* the invariant the code did not deliver. Fixed in dba837e1.

The savepoint placement was reasoned about for the wrong question. I established that it had to be a `SAVEPOINT` rather than `BEGIN` — because `append_adjudication` already holds a transaction across its call to `append` — and then never asked where it had to *open*. Outside an enclosing transaction the event insert autocommits immediately, so a savepoint opened after it protects only the edges, which is the half that was already fine.

The consequence is the one the floor exists to prevent, so it is not merely a lost index row: the surviving event has zero edges, a v7-born store's floor says `0` (fully covered), and missing edges read as *"cited nothing"*. The store would answer a provenance question confidently and wrongly, with nothing in it to contradict the answer.

It now opens before the event insert, with both writes and the unwind inside it. `SAVEPOINT` is retained for the original reason, which still holds.

I did not want to fix this on inspection alone, since inspection is what produced the bug. `a_failed_edge_write_rolls_back_the_event_row` plants an edge row at the `(generation, ordinal)` the next append will claim, so the first edge insert hits the primary key — the same window, deterministically, without simulating a crash. It asserts no `world_events` row survives. I re-applied the original placement and confirmed it reds there, then restored.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/kirra-world-store/src/lib.rs [hunk 3]
```diff
@@ -381,6 +388,7 @@ impl std::fmt::Display for StoreError {
             Self::CorruptEntityProjectionRow { detail } => {
                 write!(f, "corrupt entities_projection row: {detail}")
             }
+            Self::CitationPageSpec(e) => write!(f, "citation page: {e}"),
```
[the remark this answers is not in the document]
Comment (justinlooney) [reply crates/kirra-world-store/src/lib.rs:396]: Agreed. Took the minimal option in dba837e1: the Display message is now generic, so it no longer sends triage to a table that has nothing to do with the failure.

Worth noting the variant had already outgrown its name before this PR — it carries corrupt adjudication tokens at three existing call sites. 4a added one more use (provenance JSON), which is what made the stale message worth fixing now. The rename itself is a pre-existing cleanup across ~20 sites, and I have deliberately not smuggled it into a PR about citation edges; the reason is recorded at the variant so it does not read as an oversight.

---
_Generated by [Claude Code](https://claude.ai/code)_
